pytorch_lightning/trainer/training_io.py

```diff
@@ -401,10 +402,14 @@ def restore_training_state(self, checkpoint):
                 ' This is probably due to `ModelCheckpoint.save_weights_only` being set to `True`.'
             )
 
-        if self.checkpoint_callback is not None and self.checkpoint_callback is not False:
-            self.checkpoint_callback.best = checkpoint['checkpoint_callback_best']
+        if self.checkpoint_callback:
+            if 'checkpoint_callback_best' in checkpoint:
+                self.checkpoint_callback.best_model_score = checkpoint['checkpoint_callback_best']
+            else:
+                self.checkpoint_callback.best_model_score = checkpoint['checkpoint_callback_best_model_score']
```

Comment:
```suggestion
            # try to load the new version and if it missing use the old one            
            self.checkpoint_callback.best_model_score = checkpoint.get('checkpoint_callback_best_model_score', checkpoint.get('checkpoint_callback_best'))
```
not sure if it is not too long line

Reply:
I think it's less clear. We could keep the `if` and add a comment saying it's for checkpoints older than `0.x.y`.

Reply:
ok, then swap the logic, fist we try the new and then if it is missing try to get the old one
also add warning that you are loading old checkpoint?

Reply:
Done.